packages/core/src/2d/sprite/Sprite.ts

```diff
@@ -0,0 +1,232 @@
+import { Vector2, Vector4, Rect } from "@oasis-engine/math";
+import { RefObject } from "../../asset/RefObject";
+import { Engine } from "../../Engine";
+import { Texture2D } from "../../texture";
+
+export class Sprite extends RefObject {
+  private static _tempVec2: Vector2 = new Vector2();
+
+  private static _VERTICES_FLAG = 0x1;
+  private static _UV_FLAG = 0x2;
+  private static _TRIANGLES_FLAG = 0x4;
+  /** Sprite._VERTICES_FLAG | Sprite._UV_FLAG | Sprite._TRIANGLES_FLAG */
+  private static _VER_UV_TRI_FLAG = 0x7;
+
+  /** The array containing sprite mesh triangles. */
+  public triangles: number[] = [];
+  /** The base texture coordinates of the sprite mesh. */
+  public uv: Vector2[] = [new Vector2(), new Vector2(), new Vector2(), new Vector2()];
+  /** The array containing sprite mesh vertex positions. */
+  public vertices: Vector2[] = [new Vector2(), new Vector2(), new Vector2(), new Vector2()];
```

Comment:
"Vertex" include postion and other vertex element，if this data is just position, how about call positions